components/net/http_loader.rs

```diff
@@ -1256,8 +1256,36 @@ fn http_network_fetch(
         }
     }
 
+    let mut cloned_url = url.clone();
+    let header_strings: Vec<&str> = res
+        .headers()
+        .get_all("Timing-Allow-Origin")
+        .iter()
+        .map(|header_value| header_value.to_str().unwrap())
+        .collect();
+    let wildcard_present = header_strings
+        .iter()
+        .fold(false, |acc, header_str| acc || *header_str == "*");
+    let req_origin_in_timing_allow = header_strings
+        .iter()
+        .map(|header_string| ServoUrl::parse(header_string))
+        .filter(|header_url| header_url.is_ok())
+        .map(|header_url| header_url.unwrap().into_url())
+        .fold(false, |acc, header_url| {
```

Comment:
`.any(|header_url| header_url.origin() == url.as_url().origin())`
I don't think the cloned_url is necessary if you don't use fold.